### DISCUSSION TITLE
Retrieve package information

### DISCUSSION THREAD

**Discussion by jefersonchaves** [forcedotcom/cli#2358] [Ideas]
**Is your feature request related to a problem? Please describe.**
Today there is no simple way to retrieve package information through Salesforce CLI. There is a need to parse `sfdx-project.json` and then use two Tooling API calls to retrieve [package](https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_subscriberpackage.htm) and [package version information](https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/tooling_api_objects_subscriberpackageversion.htm) for a given alias.

**What are you trying to do**
A description of the outcome you are hoping to achieve
Retrieve package information to display in build log and to retrieve the installation key from a secret text in Jenkins. This is also useful for local development.

**Describe the solution you'd like**
Would like to have another command in `force:package` that could retrieve package information. Could be something lke `force:package:version:info`.

**Describe alternatives you've considered**
Currently, I have implemented this as a Groovy script for Jenkins but I'm also considering writing a plugin.

**Comment by fischerp** [comment]
Please assume that I have a "sfdx-project.json" file and it has a valid ID
...
"packageAliases": {
    "myPackage": "04t50000000AaaA"
}

I would like to execute the following command and receive back details about it.

Command:
`force:package:version:info -p "myPackage" --json`

Expected return:
`{"packageId":"04t50000000AaaA","name":"Package A", "majorVersion":1","minorVersion":"0"...`

**Comment by clairebianchi** [comment]
@sfdc-db-gmail please take a look

**Comment by sfdc-db-gmail** [comment]
@jefersonchaves - Can `force:package:version:list` or `force:package:version:report` provide the info you are looking for? Cc @clairebianchi

**Comment by jefersonchaves** [comment]
Hi @sfdc-db-gmail: it would work if I could retrieve information about Classic Managed Packages. My use case is: I want to retrieve information from Classic Managed Packages to use in a build, for instance.

Makes sense?

**Comment by jefersonchaves** [comment]
Hi @sfdc-db-gmail or @clairebianchi: is there anything else that I can assist?

**Comment by mshanemc** [comment]
Marking as wont-fix because (correct me if wrong) this gap only applies to 1GP.
